android/java/org/chromium/chrome/browser/playlist/settings/BravePlaylistPreferences.java

```diff
@@ -0,0 +1,201 @@
+        mStartPlaybackSwitch.setOnPreferenceChangeListener(this);
+
+        mResetPlaylist = (BravePlaylistResetPreference) findPreference(PREF_RESET_PLAYLIST);
+        mResetPlaylist.setOnPreferenceClickListener(preference -> {
+            if (mPlaylistService != null) {
+                mPlaylistService.resetAll();
+                PlaylistPreferenceUtils.resetPlaylistPrefs(getActivity());
+                BraveRelaunchUtils.askForRelaunch(getActivity());
+            }
+            return true;
+        });
+
+        updatePlaylistSettingsState(
+                SharedPreferencesManager.getInstance().readBoolean(PREF_ENABLE_PLAYLIST, true));
+    }
+
+    @Override
+    public void onDestroy() {
+        if (mPlaylistService != null) {
+            mPlaylistService.close();
+        }
+        super.onDestroy();
+    }
+
+    @Override
+    public void onConnectionError(MojoException e) {
+        mPlaylistService = null;
+        initPlaylistService();
+    }
+
+    private void initPlaylistService() {
+        if (mPlaylistService != null) {
+            return;
+        }
+
+        mPlaylistService = PlaylistServiceFactoryAndroid.getInstance().getPlaylistService(this);
+    }
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        getActivity().setTitle(R.string.brave_playlist);
+        SettingsUtils.addPreferencesFromResource(this, R.xml.brave_playlist_preferences);
+        initPlaylistService();
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        new Handler().post(() -> { updateAutoSaveMedia(); });
+    }
+
+    private void updateAutoSaveMedia() {
+        if (mAutoSaveMediaForOfflinePreference != null) {
+            switch (SharedPreferencesManager.getInstance().readInt(
+                    PREF_AUTO_SAVE_MEDIA_FOR_OFFLINE, 0)) {
+                case 0:
+                    mAutoSaveMediaForOfflinePreference.setSummary(
+                            getActivity().getResources().getString(R.string.on_text));
+                    break;
+                case 1:
+                    mAutoSaveMediaForOfflinePreference.setSummary(
+                            getActivity().getResources().getString(R.string.off_text));
+                    break;
+                case 2:
+                    mAutoSaveMediaForOfflinePreference.setSummary(
+                            getActivity().getResources().getString(R.string.wifi_only_text));
+                    break;
+            }
+        }
+    }
+
+    @Override
+    public boolean onPreferenceChange(Preference preference, Object newValue) {
+        String key = preference.getKey();
+        if (PREF_ENABLE_PLAYLIST.equals(key)) {
+            SharedPreferencesManager.getInstance().writeBoolean(
+                    PREF_ENABLE_PLAYLIST, (boolean) newValue);
+            updatePlaylistSettingsState((boolean) newValue);
+            BraveRelaunchUtils.askForRelaunch(getActivity());
```

Comment:
Do you know , does desktop also needs to be relaunched after enabling playlists?
I have this question, because usually relaunch is required when flag for feature is clanged which is read during process launch, and I don't see flags here.